specs/driver_spec.rb

```diff
@@ -0,0 +1,192 @@
+        first_driver.length.must_equal 1
+      end
+
+      it "returns empty array if the driver does not exist" do
+        imaginary_driver.must_be_empty
+      end
+    end
+
+    describe "subset_driver_trips" do
+      let(:first_driver_trips) {Rideshare::Driver.subset_driver_trips(1)}
+
+      let(:last_driver_trips) {Rideshare::Driver.subset_driver_trips(99)}
+
+
+      it "returns the correct trips for the first driver" do
+
+        first_driver_trips.length.must_equal 9
+
+        trip_ids = []
+        first_driver_trips.each do |trip|
+          trip_ids << trip.trip_id.to_i
+        end
+        trip_ids.sort.must_equal [1, 122, 124, 216, 417, 434, 439, 530, 553]
+      end
+
+
+      it "returns the correct trips for the last driver" do
+
+        last_driver_trips.length.must_equal 6
+
+        trip_ids = []
+        last_driver_trips.each do |trip|
+          trip_ids << trip.trip_id.to_i
+        end
+        trip_ids.sort.must_equal [54, 86, 230, 420, 423, 531]
+      end
+
+    end
+
+
+    describe "get_driver_rating" do
+
+      let(:first_driver_rating) {Rideshare::Driver.get_driver_rating(1)}
+
+      let(:last_driver_rating) {Rideshare::Driver.get_driver_rating(99)}
+
+      let(:non_driving_rating) { Rideshare::Driver.get_driver_rating(100)}
+
+      let(:rating_too_high){Rideshare::Driver.driver_rating([
+        Rideshare::Trip.new({ trip_id: 1,driver_id: 1,rider_id: 1,date: "today",rating: 6 }, :driver_id),
+
+        Rideshare::Trip.new( { trip_id: 2,driver_id: 2,rider_id: 2,date: "today",rating: 6 }, :driver_id ),
+
+        Rideshare::Trip.new( { trip_id: 3,driver_id: 3,rider_id: 3,date: "today",rating: 3 }, :driver_id),
+
+        Rideshare::Trip.new( { trip_id: 4,driver_id: 4,rider_id: 4,date: "today",rating: 3 }, :driver_id)
+        ])}
+
+        let(:rating_too_low){Rideshare::Driver.driver_rating([
+          Rideshare::Trip.new({ trip_id: 1,driver_id: 1,rider_id: 1,date: "today",rating: 0 }, :driver_id),
+
+          Rideshare::Trip.new( { trip_id: 2,driver_id: 2,rider_id: 2,date: "today",rating: 0 }, :driver_id ),
+
+          Rideshare::Trip.new( { trip_id: 3,driver_id: 3,rider_id: 3,date: "today",rating: 3 }, :driver_id),
+
+          Rideshare::Trip.new( { trip_id: 4,driver_id: 4,rider_id: 4,date: "today",rating: 3 }, :driver_id)
+          ])}
+
+          let(:only_weird_rating){Rideshare::Driver.driver_rating([Rideshare::Trip.new( { trip_id: 4,driver_id: 4,rider_id: 4,date: "today",rating: 6 }, :driver_id ) ] ) }
+
+            it "correctly calculates the average rating for the first driver based on all trips taken" do
+              first_driver_rating.must_equal 2.6
+            end
+
+            it "correctly calculates the average rating for the last driver based on all trips taken" do
+              last_driver_rating.must_equal 2.8
+            end
+
+            it "returns nil and outputs a warning if the driver has not done any properly rated trips yet" do
```

Comment:
Good test case! Edge cases like this, asking what happens when there *isn't* any data, can be easy to miss.